server-spi-private/src/main/java/org/keycloak/migration/migrators/MigrateTo18_0_0.java

```diff
@@ -0,0 +1,52 @@
+package org.keycloak.migration.migrators;
+
+import java.util.Objects;
+
+import org.keycloak.migration.ModelVersion;
+import org.keycloak.models.AccountRoles;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RoleModel;
+import org.keycloak.representations.idm.RealmRepresentation;
+
+import static org.keycloak.models.AccountRoles.MANAGE_ACCOUNT;
+import static org.keycloak.models.AccountRoles.MANAGE_ACCOUNT_2FA;
+import static org.keycloak.models.AccountRoles.MANAGE_ACCOUNT_BASIC_AUTH;
```

Comment:
I'd rather have a single role "MANAGE_CREDENTIALS" than these two.

Reply:
When user is from external IdP, he must not be able to add Keycloak password. However, he must be able to add two factor credentials. Can we block external users from login via Keycloak Username form in another way?